engine/spverrors/definitions.go

```diff
@@ -0,0 +1,338 @@
+package spverrors
+
+import "github.com/bitcoin-sv/spv-wallet/models"
+
+// How the Codes are generated?
+// 1. The first part of the code is the error prefix, e.g. "error"
+// 2. The second part of the code is the model name, e.g. "transaction"
+// 2. The third part of the code is the reason of the error, e.g. "not-found"
```

Comment:
This description doesn't align to what I see below.
Maybe sth like this:
- `"error"` - like mandatory prefix for all error codes
- (optional) `{error group}` -  e.g. `"unauthorized"`
- (optional) `{subject}` - name of model or some noun e.g. `body`, `auth-header`
- (optional) `{reason}` - what happened, e.g. `"not-found"`, `"missing"`, `"invalid"`

So the error codes would look like this:
- `error-unauthorized`
- `error-unauthorized-token-missing`
- `error-unauthorized-signature-invalid`

Reply:
Done